lib/bannerFactory.js

```diff
@@ -33,21 +35,83 @@ class BannerFactory extends BaseFactory {
      * @param {String}  config.createdBy           The username of the associated user
      * @param {Boolean} [config.isActive=false]    Whether the banner is active
      * @param {String}  [config.type='info']       Type of banner (info|warn|etc)
+     * @param {String}  [config.scope='GLOBAL']    Scope of the banner (GLOBAL|PIPELINE|BUILD)
      * @memberof BannerFactory
      */
-    create(config) {
+    async create(config) {
         if (!config.type) {
             config.type = 'info';
         }
         if (!config.isActive) {
             config.isActive = false;
         }
+        if (!config.scope) {
+            config.scope = 'GLOBAL';
+            config.scopeId = null;
+        }
+        if ((config.scope === 'PIPELINE' || config.scope === 'BUILD') && !config.scopeId) {
+            throw new Error(`scopeId is required when scope is ${config.scope}`);
+        }
+        if (config.scope === 'PIPELINE') {
+            const pipeline = await PipelineFactory.getInstance().get(config.scopeId);
+
+            if (!pipeline) {
+                throw new Error(`Pipeline ${config.scopeId} does not exist`);
+            }
+        }
+        if (config.scope === 'BUILD') {
+            const build = await BuildFactory.getInstance().get(config.scopeId);
+
+            if (!build) {
+                throw new Error(`Build ${config.scopeId} does not exist`);
+            }
+        }
 
         config.createTime = new Date().toISOString();
 
         return super.create(config);
     }
 
+    /**
+     * Helper function to ensure scope and scopeId are set
+     * @param {Object} banner
+     * @return {Object} banner
+     */
+    _setScopeDefaults(banner) {
+        if (!banner.scope) {
+            banner.scope = 'GLOBAL';
+        }
+        if (!banner.scopeId) {
+            banner.scopeId = null;
+        }
+
+        return banner;
+    }
+
+    /**
+     * Retrieves a banner by its ID and sets default scope values.
+     *
+     * @param {number|string} id - The ID of the banner to retrieve.
+     * @returns {Promise<Object>} A promise that resolves to the banner object with default scope values set.
+     */
+    async get(id) {
+        const response = await super.get(id);
+
+        return this._setScopeDefaults(response);
+    }
+
+    /**
+     * Retrieves a list of banners and sets default scope values for each banner.
+     *
+     * @param {Object} config - The configuration object for the list request.
+     * @returns {Promise<Array>} A promise that resolves to an array of banners with default scope values set.
+     */
+    async list(config) {
+        const response = await super.list(config);
+
+        return response.map(banner => this._setScopeDefaults(banner));
+    }
+
```

Comment:
I believe we don't need this. Once the DB schema changes are applied, the defaults should be set in the DB for all the existing records.
```suggestion
```